src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6968,25 +6979,50 @@ private bool InEnumMemberInitializer()
             return new BoundPropertyAccess(node, receiver, propertySymbol, lookupResult, propertySymbol.Type, hasErrors: (hasErrors || hasError));
         }
 
-        private void CheckRuntimeSupportForSymbolAccess(SyntaxNode node, BoundExpression receiverOpt, Symbol symbol, BindingDiagnosticBag diagnostics)
+        private void CheckReceiverAndRuntimeSupportForSymbolAccess(SyntaxNode node, BoundExpression receiverOpt, Symbol symbol, BindingDiagnosticBag diagnostics)
         {
-            if (symbol.ContainingType?.IsInterface == true && !Compilation.Assembly.RuntimeSupportsDefaultInterfaceImplementation && Compilation.SourceModule != symbol.ContainingModule)
+            if (symbol.ContainingType?.IsInterface == true)
             {
-                if (!symbol.IsStatic && !(symbol is TypeSymbol) &&
-                    !symbol.IsImplementableInterfaceMember())
+                if (symbol.IsStatic && symbol.IsAbstract)
                 {
-                    Error(diagnostics, ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, node);
+                    Debug.Assert(symbol is not TypeSymbol);
+
+                    if (receiverOpt is BoundQueryClause { Value: var value })
+                    {
+                        receiverOpt = value;
+                    }
+
+                    if (receiverOpt is not BoundTypeExpression { Type: { TypeKind: TypeKind.TypeParameter } })
+                    {
+                        Error(diagnostics, ErrorCode.ERR_BadAbstractStaticMemberAccess, node);
+                        return;
+                    }
+
+                    if (!Compilation.Assembly.RuntimeSupportsStaticAbstractMembersInInterfaces && Compilation.SourceModule != symbol.ContainingModule)
```

Comment:
>Compilation.SourceModule != symbol.ContainingModule [](start = 98, length = 51)

Am I correct that the goal of this is to minimize errors? Things declared in source will have an error on declaration, we only want to error on usages that don't have their own error?

Reply:
> Am I correct that the goal of this is to minimize errors? Things declared in source will have an error on declaration, we only want to error on usages that don't have their own error?

That is correct, the goal is to reduce the noise at a consumption site when we are going to complain at the declaration site anyway.

---
In reply to: [613564070](https://github.com/dotnet/roslyn/pull/52605#discussion_r613564070) [](ancestors = 613564070)